Branch-SDK/src/androidTest/java/io/branch/referral/network/BranchAsyncNetworkLayerTest.kt

```diff
@@ -0,0 +1,162 @@
+package io.branch.referral.network
+
+import io.branch.referral.Branch
+
+/**
+ * Simple tests for BranchAsyncNetworkLayer.
+ * 
+ * These tests verify basic functionality without external test framework dependencies.
+ * They focus on testing the Thread.sleep() elimination and coroutine-based implementation.
+ */
+class BranchAsyncNetworkLayerTest {
+    
+    /**
+     * Test that BranchAsyncNetworkLayer can be instantiated successfully.
+     */
+    fun testAsyncNetworkLayerInstantiation() {
+        try {
+            val mockBranch = createMockBranch()
+            val networkLayer = BranchAsyncNetworkLayer(mockBranch)
+            
+            // If we get here without exception, the test passes
+            println("✓ BranchAsyncNetworkLayer instantiation test passed")
+        } catch (e: Exception) {
+            println("✗ BranchAsyncNetworkLayer instantiation test failed: ${e.message}")
+            throw e
+        }
+    }
+    
+    /**
+     * Test that the network layer can be cancelled without throwing exceptions.
+     */
+    fun testNetworkLayerCancellation() {
+        try {
+            val mockBranch = createMockBranch()
+            val networkLayer = BranchAsyncNetworkLayer(mockBranch)
+            
+            // Test cancellation
+            networkLayer.cancelAll()
+            
+            println("✓ Network layer cancellation test passed")
+        } catch (e: Exception) {
+            println("✗ Network layer cancellation test failed: ${e.message}")
+            throw e
+        }
+    }
+    
+    /**
+     * Test that multiple network layers can be created independently.
+     */
+    fun testMultipleNetworkLayerInstances() {
+        try {
+            val mockBranch = createMockBranch()
+            val networkLayer1 = BranchAsyncNetworkLayer(mockBranch)
```

Comment:
This is to test that multiple network requests can execute concurrently correct?